includes/upload/UploadBase.php

```diff
@@ -249,6 +251,25 @@ public function verifyUpload() {
 			);
 		}
 
+		// Wikia change - check if image resolution does not exceed wgMaxImageArea - @author: ryba
+		$imgInfo = getimagesize($this->mUpload->getTempName());
+		$imgWidth = $imgInfo[0];
+		$imgHeight = $imgInfo[1];
+		$mime = $imgInfo['mime'];
+		$imageResolution = $imgHeight * $imgWidth;
+		if ( $imageResolution > $wgMaxImageArea && $mime !== 'image/jpeg' ) {
```

Comment:
It would be cool to have consistent rules across all file types, however if we introduce it now, thumbnails of jpeg images larger than wgMaxImageArea (that are already uploaded) will stop rendering on a page